src/iterators.rs

```diff
@@ -482,6 +484,61 @@ pub struct OuterIter<'a, A: 'a, D> {
     life: PhantomData<&'a A>,
 }
 
+impl<'a, A, D> OuterIter<'a, A, D>
+    where D: Dimension
+{
+    /// Split the iterator at index, yielding two disjoint iterators.
+    ///
+    /// *panics* if `index` is strictly greater than the iterator's length
+    pub fn split_at(self, index: Ix)
+        -> (OuterIter<'a, A, D>, OuterIter<'a, A, D>)
+    {
+        assert!(index <= self.iter.len);
+        let right_ptr = unsafe { self.iter.offset(index) };
```

Comment:
Ah how annoying, we end up with the same pointer past the end trouble here. Needs to handle the == len case differently.